### DISCUSSION TITLE
Is it necessary to sleep in a loop?

### DISCUSSION THREAD

**Discussion by leecallen35** [micropython/micropython#15834] [RP2040 / Pico]
Sorry I realize this is a noob question. On any other platform I would put a sleep in a loop so the program does not monopolize the CPU. Is that necessary & appropriate in micropython on the Pico?  

Thanks.

**Comment by robert-hh** [comment]
That is generally useful if you application can accepts a delay up to 1 ms. Best call machine.idle().

**Reply by leecallen35** [reply]
Thank you! Fast and helpful!
